ts-discordjs/index.ts

```diff
@@ -0,0 +1,57 @@
+
+/**
+ * @klotho::persist {
+ *   id = "secret"
+ *   secret = true
+ * }
+ */
+import * as fs from "fs/promises"
+
+import { Client, GatewayIntentBits, RESTPutAPIApplicationGuildCommandsResult, Routes, SlashCommandBuilder } from "discord.js"
+
+interface discordConfig {
+  token: string,
+  clientId: string,
+  guildId: string,
+}
+
+const config = async function() {
+  return JSON.parse(await fs.readFile("discord_config.json", 'utf-8')) as discordConfig
+}()
+
+const client = async function() {
+  const client = new Client({intents: [GatewayIntentBits.Guilds]})
+
+  client.once('ready', () => console.info("Bot ready"))
+
+  client.login((await config).token)
+
+  return client
+}()
+
+const commands = [
+  new SlashCommandBuilder().setName('ping').setDescription('Replies with pong1'),
+].map(c => c.toJSON())
+
+client.then(async c => {
+  const cfg = await config
+  try {
+    const data = await c.rest.put(Routes.applicationGuildCommands(cfg.clientId, cfg.guildId), {body: commands}) as RESTPutAPIApplicationGuildCommandsResult
+    console.info(`Successfully registered ${data.length} application commands.`)
```

Comment:
minor: consider a ternary for only pluralizing "commands" only if there are != 1 of them.